src/server/game/Spells/SpellMgr.cpp

```diff
@@ -3070,10 +3070,11 @@ void SpellMgr::LoadSpellInfoCustomAttributes()
                                     if (spellInfo->SpellFamilyName != SPELLFAMILY_MAGE || !(spellInfo->SpellFamilyFlags[0] & 0x20)) // frostbolt
                                         if (spellInfo->Id != 55095) // frost fever
                                             if (spellInfo->SpellFamilyName != SPELLFAMILY_WARLOCK || !(spellInfo->SpellFamilyFlags[1] & 0x40000)) // Haunt
-                                            {
-                                                spellInfo->AttributesCu |= SPELL_ATTR0_CU_BINARY_SPELL;
-                                                break;
-                                            }
+                                                if (spellInfo->SpellFamilyName != SPELLFAMILY_WARLOCK || !(spellInfo->SpellFamilyFlags[0] & 0x4000)) // Drain Soul
```

Comment:
its being removed from haunt but added to drain soul?

Reply:
Nooo, please, no more inverted nested ifs, I can't take it anymore 😭
This code block is unreadable.

Can we rewrite it to something like this?
```
if (!spellInfo->Effects[j].CalcValue() && 
    !(spellInfo->Effects[j].Effect == SPELL_EFFECT_INTERRUPT_CAST || spellInfo->HasAttribute(SPELL_ATTR0_CU_DONT_BREAK_STEALTH)) || 
     spellInfo->HasAttribute(SPELL_ATTR0_NO_IMMUNITIES))
{
    continue;
}

if (spellInfo->Id == 69649 || spellInfo->Id == 71056 || spellInfo->Id == 71057 || spellInfo->Id == 71058 ||
    spellInfo->Id == 73061 || spellInfo->Id == 73062 || spellInfo->Id == 73063 || spellInfo->Id == 73064)
{
    continue;
}

if (spellInfo->SpellFamilyName == SPELLFAMILY_MAGE && (spellInfo->SpellFamilyFlags[0] & 0x20)) // Frostbolt
    continue;

if (spellInfo->Id == 55095) // Frost Fever
    continue;

if (spellInfo->SpellFamilyName == SPELLFAMILY_WARLOCK &&
    ((spellInfo->SpellFamilyFlags[1] & 0x40000) || (spellInfo->SpellFamilyFlags[0] & 0x4000))) // Haunt or Drain Soul
{
    continue;
}

// your condition

spellInfo->AttributesCu |= SPELL_ATTR0_CU_BINARY_SPELL;
break;
```